pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -358,6 +347,22 @@ export async function resolveRootDependencies (
       }
       pkgAddresses.push(...importerResolutionResult.pkgAddresses)
     }
+    if (Object.keys(allMissingOptionalPeers).length && ctx.allPreferredVersions) {
+      const optionalDependencies = hoistOptionalPeers(allMissingOptionalPeers, ctx.allPreferredVersions)
+      if (Object.keys(optionalDependencies).length) {
+        const wantedDependencies = getNonDevWantedDependencies({ optionalDependencies })
+        const resolveDependenciesResult = await resolveDependencies(ctx, preferredVersions, wantedDependencies, {
+          ...options,
+          parentPkgAliases,
+          publishedBy,
+        })
+        importerResolutionResult = {
+          pkgAddresses: resolveDependenciesResult.pkgAddresses,
+          ...filterMissingPeers(await resolveDependenciesResult.resolvingPeers, parentPkgAliases),
+        }
+        pkgAddresses.push(...importerResolutionResult.pkgAddresses)
```

Comment:
I am not sure what you mean. How does an array concatenation cause a stackoverflow?

Reply:
Sorry, I should have cited something. The best discussion of the issue I can find is:

https://github.com/nodejs/node/issues/16870

The problem is that all arguments are placed onto the stack one by one before calling `.push`, which means that this code can throw depending on how deep in the stack we are already and argument size.

I tend to avoid spreading arguments into a function if the arguments aren't a fixed size so I don't have to think about this problem.

Reply:
oh wow, good to know